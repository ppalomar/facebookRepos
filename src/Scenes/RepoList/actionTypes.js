export const REPOS_FETCH = 'REPOS_FETCH';
export const REPOS_FETCH_SUCCESS = 'REPOS_FETCH_SUCCESS';

export const REPO_GO_DETAIL = 'REPO_GO_DETAIL';
