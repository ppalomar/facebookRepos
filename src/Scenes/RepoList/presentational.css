.repos-list__container{
    display: flex;
}

.repos-list__container .left{
    flex:1;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    border-right: 1px #2196F3 solid;
}

.repos-list__container .right{
    flex: 4;
}
