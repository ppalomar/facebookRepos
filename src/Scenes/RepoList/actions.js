import * as AT from './actionTypes';

export function fetchRepos() {
  return {
    type: AT.REPOS_FETCH,
  };
}

export function fetchReposSuccess(payload) {
  return {
    type: AT.REPOS_FETCH_SUCCESS,
    repos: payload.data
  };
}

export function fetchReposFailed(error) {
  console.error(error);
}

export function goRepoDetail(repo) {
  return {
    type: AT.REPO_GO_DETAIL,
    repo,
  };
}
