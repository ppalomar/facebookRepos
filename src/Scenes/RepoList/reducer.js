import * as AT from './actionTypes';

export const INITIAL_STATE = {
  loading: false,
  repos: [],
  selectedRepo: undefined,
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.REPOS_FETCH:
      return {
        ...state,
        loading: true,
      };
    case AT.REPOS_FETCH_SUCCESS:
      return {
        ...state,
        loading: false,
        repos: action.repos,
      };
    case AT.REPO_GO_DETAIL:
      return {
        ...state,
        selectedRepo: action.repo,
      };
    default:
      return state;
  }
};
