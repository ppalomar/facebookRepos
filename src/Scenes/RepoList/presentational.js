import React, { Fragment } from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';

import RepoRow from './components/RepoRow';

import './presentational.css';

const RepoListStateless = ({ loading, repos, goDetail }) => {
  const isEmpty = repos.length === 0;

  const list = (
    <Fragment>
      {repos.map(({ name }) => (
        <RepoRow key={name} repoName={name} goDetail={goDetail} />
      ))}
    </Fragment>
  );

  return (
    <div className="repos-list__container">
      <div className="left">
        {loading && isEmpty ? <CircularProgress size={50} /> : list}
      </div>
      <div className="right">
      </div>
    </div>
  );
};

export default RepoListStateless;
