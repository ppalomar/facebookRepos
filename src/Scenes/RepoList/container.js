import { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { fetchRepos, goRepoDetail } from './actions';
import { getSortedReposSelector } from './selectors';

import StatelessComponent from './presentational';

const stateToProps = (state, { history, match }) => ({
  loading: state.Scenes.RepoList.loading,
  history,
  repos: getSortedReposSelector(state),
  seletedRepo: match.params && match.params.repoName,
});

const dispatchToProps = dispatch => ({
  onLoad: () => dispatch(fetchRepos()),
  goDetail: repoName => dispatch(goRepoDetail(repoName)),
  onShowMore: since => dispatch(fetchRepos({ since }, true)),
});

const mergeProps = (stateProps, dispatchProps) => ({
  ...stateProps,
  ...dispatchProps,
  goDetail: repoName => {
    const { history } = stateProps;
    history.push(`${repoName}`);
    dispatchProps.goDetail(repoName);
  },
});

class ReposList extends Component {
  componentDidMount() {
    this.props.onLoad();
  }

  render() {
    return StatelessComponent(this.props);
  }
}

ReposList.propTypes = {
  ...StatelessComponent.propTypes,
  onLoad: PropTypes.func.isRequired,
};

export default connect(stateToProps, dispatchToProps, mergeProps)(ReposList);
