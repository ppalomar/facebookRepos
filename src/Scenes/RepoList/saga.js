import { call, put, takeLatest } from 'redux-saga/effects';

import { getRepos as API_fetchRepos } from 'API/repos';
import * as Actions from './actions';
import { REPOS_FETCH } from './actionTypes';

function* fetchRepos() {
  try {
    const payload = yield call(API_fetchRepos);
    yield put(Actions.fetchReposSuccess(payload));
  } catch (e) {
    yield put(Actions.fetchReposFailed(e));
  }
}

function* saga() {
  yield takeLatest(REPOS_FETCH, fetchRepos);
}

export default saga;
