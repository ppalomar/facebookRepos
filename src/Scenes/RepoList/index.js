import RepoList from './container';
import saga from './saga';

export { RepoList, saga };
export default RepoList;
