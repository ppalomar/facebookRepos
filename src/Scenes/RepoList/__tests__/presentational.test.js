import React from 'react';
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button';

import RepoList from '../presentational';

describe('RepoRow', () => {
  it('renders progress component when loading is true', () => {
    const props = {
      loading: true,
      repos: [],
      goDetail: () => {},
      onShowMore: () => {},
    };

    const wrapper = shallow(<RepoList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders list of repos when loading is false', () => {
    const props = {
      loading: false,
      repos: [
        {
          name: 'Pablo',
        },
        {
          name: 'Cristina',
        },
      ],
      goDetail: () => {},
      onShowMore: () => {},
    };

    const wrapper = shallow(<RepoList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
