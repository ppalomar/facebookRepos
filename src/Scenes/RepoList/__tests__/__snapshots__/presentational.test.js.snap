// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoRow renders list of repos when loading is false 1`] = `
<div
  className="repos-list__container"
>
  <div
    className="left"
  >
    <React.Fragment>
      <RepoRow
        goDetail={[Function]}
        key="Pablo"
        repoName="Pablo"
      />
      <RepoRow
        goDetail={[Function]}
        key="Cristina"
        repoName="Cristina"
      />
    </React.Fragment>
  </div>
  <div
    className="right"
  />
</div>
`;

exports[`RepoRow renders progress component when loading is true 1`] = `
<div
  className="repos-list__container"
>
  <div
    className="left"
  >
    <WithStyles(CircularProgress)
      size={50}
    />
  </div>
  <div
    className="right"
  />
</div>
`;
